registration/include/pcl/registration/impl/gicp.hpp

```diff
@@ -43,6 +43,18 @@
 #include <pcl/registration/boost.h>
 #include <pcl/registration/exceptions.h>
 
+#include <algorithm>
+namespace std {
+#ifndef clamp
+template<class T>
+constexpr const T& clamp( const T& v, const T& lo, const T& hi )
```

Comment:
Wrong place (file and namespace) for this function. There's already a ton of usage of `std::min(std::max(v, lo), hi)` in PCL codebase. Also, modification of std namespace is frowned upon.

To check for clamp, use `__cpp_lib_clamp` as the standard macro

Reply:
@kunaltyagi : ok, just corrected that point.